lib/msal-node/test/client/ConfidentialClientApplication.spec.ts

```diff
@@ -340,6 +343,87 @@ describe("ConfidentialClientApplication", () => {
             );
         });
 
+        describe("region is determined correctly", () => {
+            let acquireTokenByClientCredentialSpy: jest.SpyInstance;
+            let buildOauthClientConfigurationSpy: jest.SpyInstance;
+            let client: ConfidentialClientApplication;
+            let request: ClientCredentialRequest;
+            beforeEach(() => {
+                acquireTokenByClientCredentialSpy = jest.spyOn(
```

Comment:
Why not use HTTP mocking and assert that the token request goes over the ESTS-R token endpoint?